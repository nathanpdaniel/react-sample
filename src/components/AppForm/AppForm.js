import React from 'react';
import { Field, reduxForm } from 'redux-form';
import classnames from 'classnames';
import { mustBeWholeNumber, mustBeGreaterThan, mustBeNumeric } from '../../utils/validators';
import './AppForm.scss';

const mustBeGreaterThan2 = mustBeGreaterThan(2);

const renderField = ({ input, label, type, meta: { touched, error, warning } }) => (
  <div>
    {touched && ((error && <span>{error}</span>) || (warning && <span>{warning}</span>))}
    <input className={classnames({ error: touched && error })} {...input} placeholder={label} type={type} />
  </div>
)

export const AppForm = (props) => (
  <form className={classnames('appForm')} onSubmit={props.handleSubmit}>
    <div>
      <label>Sample Size:</label>
      <div>
        <Field
          name="sampleSize"
          component={renderField}
          type="text"
          placeholder="Sample Size"
          validate={[mustBeNumeric, mustBeWholeNumber, mustBeGreaterThan2]}
        />
      </div>
    </div>
    <div>
      <label>Sample Mean:</label>
      <div>
        <Field
          name="sampleMean"
          component={renderField}
          type="text"
          placeholder="Sample Mean"
          validate={[mustBeNumeric]}
        />
      </div>
    </div>
    <div>
      <label>Standard Deviation:</label>
      <div>
        <Field
          name="standardDeviation"
          component={renderField}
          type="text"
          placeholder="Standard Deviation"
          validate={[mustBeNumeric]}
        />
      </div>
    </div>
    <div>
      <div className={classnames('checkbox')}>
        <Field name="testHypothesis" component="input" type="checkbox" />
        <label htmlFor="testHypothesis">Perform Hypothesis Test</label>
      </div>
    </div>
    <div className={classnames({ variable: true, disabled: !props.performHypothesisTest })}>
      <label>Hypothesized Mean:</label>
      <div>
        <Field
          name="hypothesizedMean"
          component={renderField}
          type="text"
          placeholder="Hypothesized Mean"
          disabled={props.performHypothesisTest}
          validate={[mustBeNumeric]}
        />
      </div>
    </div>
    <div className={classnames('buttons')}>
      <button type="submit" disabled={props.submitting}>Ok</button>
      <button type="reset" disabled={props.submitting} onClick={props.reset}>Reset</button>
    </div>
  </form>
);

export default reduxForm({
  form: 'appForm'
})(AppForm);