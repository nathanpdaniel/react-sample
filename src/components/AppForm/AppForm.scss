.appForm {
  display: flex;
  flex-direction: column;
  padding: 24px;

  > div {
    display: flex;
    width: 100%;
    padding-bottom: 18px;

    label {
      width: 25%;
      font-size: 12px;
    }

    div {
      width: 100%;

      span {
        font-size: 11px;
        color: red;
      }

      input {
        width: calc(100% - 24px);
        font-size: 12px;
        padding: 8px 12px;

        border: 1px solid #ccc;
        border-radius: 4px;

        &.error {
          border: 1px solid red;
        }
      }

      &.checkbox {
        input {
          width: 24px;
        }
      }
    }

    &.variable {
      label {
        padding-left: 48px;
        width: calc(25% - 48px);
      }

      &.disabled {
        opacity: .5;
      }
    }

    &.buttons {
      display: flex;
      justify-content: flex-end;

      button {
        border-radius: 4px;
        margin: 0 8px;
        min-width: 80px;
        padding: 10px 20px;


        &[type=submit] {
          color: white;
          background-color: rgba(0, 85, 255, 1);
          border: 1px solid  rgba(0, 85, 255, 1);

          &:hover {
            background-color: rgba(0, 85, 255, .5);
          }
        }

        &[type=reset] {
          color: black;
          background-color: rgba(200, 200, 200, 1);
          border: 1px solid darkgray;

          &:hover {
            background-color: rgba(200, 200, 200, .5);
          }
        }
      }
    }
  }
}