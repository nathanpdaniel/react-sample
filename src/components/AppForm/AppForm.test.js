import React from 'react';
import { shallow } from 'enzyme';
import AppForm from './AppForm';

describe('AppForm Tests', () => {
  it('should display properly', () => {
    const component = shallow(<AppForm />);
    expect(component).toMatchSnapshot();
  });
});