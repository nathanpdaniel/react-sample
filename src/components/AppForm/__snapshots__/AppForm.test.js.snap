// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppForm Tests should display properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef
    destroyOnUnmount={true}
    enableReinitialize={false}
    forceUnregisterOnUnmount={false}
    form="appForm"
    getFormState={[Function]}
    keepDirtyOnReinitialize={false}
    persistentSubmitErrors={false}
    pure={true}
    shouldAsyncValidate={[Function]}
    shouldError={[Function]}
    shouldValidate={[Function]}
    shouldWarn={[Function]}
    touchOnBlur={true}
    touchOnChange={false}
    updateUnregisteredFields={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "destroyOnUnmount": true,
      "enableReinitialize": false,
      "forceUnregisterOnUnmount": false,
      "form": "appForm",
      "forwardedRef": null,
      "getFormState": [Function],
      "keepDirtyOnReinitialize": false,
      "persistentSubmitErrors": false,
      "pure": true,
      "shouldAsyncValidate": [Function],
      "shouldError": [Function],
      "shouldValidate": [Function],
      "shouldWarn": [Function],
      "touchOnBlur": true,
      "touchOnChange": false,
      "updateUnregisteredFields": false,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "destroyOnUnmount": true,
        "enableReinitialize": false,
        "forceUnregisterOnUnmount": false,
        "form": "appForm",
        "forwardedRef": null,
        "getFormState": [Function],
        "keepDirtyOnReinitialize": false,
        "persistentSubmitErrors": false,
        "pure": true,
        "shouldAsyncValidate": [Function],
        "shouldError": [Function],
        "shouldValidate": [Function],
        "shouldWarn": [Function],
        "touchOnBlur": true,
        "touchOnChange": false,
        "updateUnregisteredFields": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
